### PR TITLE
Update dependencies and move to ESM

### PR DESCRIPTION
**Pull request by pioug**
Thanks for releasing a [new version](https://github.com/sindresorhus/vinyl-file/releases/tag/v5.0.0) of `vinyl-file` 🙏 This time, I want to update `gulp-rev`!

### What's new?

- Update all dependencies
- Migrate to ESM
- Migrate async `ava` tests using `p-event`
- Fix linting errors with new `xo`

[The tests are passing in my repo](https://github.com/pioug/gulp-rev/actions). 

I also updated some examples in the readme but I am not too sure about the ones that mix ESM and CJS modules.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks. Would you be willing to move to use `easy-transform-stream` too? See: https://github.com/sindresorhus/gulp-strip-debug/commit/a8d44b630bd12f2004bb0fc5d0178ecf756a128d

**Comment by pioug** [comment]
@sindresorhus Thanks for the suggestion! I didn't know about this package. I'll update the PR again over the weekend.

File: readme.md [hunk 1]
```diff
@@ -83,18 +83,18 @@ The hash of each rev'd file is stored at `file.revHash`. You can use this for cu
 ### Asset manifest
 
 ```js
-const gulp = require('gulp');
-const rev = require('gulp-rev');
+import {src, dest} from 'gulp';
```
Comment (sindresorhus) [inline readme.md:86]: I would prefer to keep `gulp` a default import.

File: readme.md [hunk 2]
```diff
@@ -136,18 +136,18 @@ Because of the way `gulp-concat` handles file paths, you may need to set `cwd` a
...
 ```js
 const gulp = require('gulp');
-const rev = require('gulp-rev');
 const sourcemaps = require('gulp-sourcemaps');
 const concat = require('gulp-concat');
 
-exports.default = () => (
-	gulp.src('src/*.js')
+exports.default = async () => {
+	const {default: rev} = await import('gulp-rev');
```
Comment (sindresorhus) [inline readme.md:143]: Why dynamic import?

File: test/rev.js [hunk 3]
```diff
@@ -44,72 +44,78 @@ test('stores the hashes for later', async t => {
...
 		path: 'pastissada.css.map',
-		contents: JSON.stringify({})
+		contents: JSON.stringify({}),
 	}));
+
+	for await (const file of data) {
+		if (path.extname(file.path) === '.map') {
+			t.is(file.path, 'pastissada-d41d8cd98f.css.map');
+		}
+	}
```
Comment (sindresorhus) [inline test/rev.js:110]: Do an assert for the correct file count too. Currently, if the stream never emits, it would not be caught and test would succeed.

**Comment by sindresorhus** [comment]
> I migrated to easy-transform-stream but I have a problem with the flush function on Node 14 (tests are passing with Node 15+) 🤔

I think we can just target Node.js 16. This is a dev tool anyway, not a reusable package.

File: readme.md (hunk 1 above)
Comment (pioug) [reply readme.md]: ➡️ https://github.com/sindresorhus/gulp-rev/pull/267/commits/39e09fa8de35869791c9942f33c444927079a1c7

File: readme.md (hunk 2 above)
Comment (pioug) [reply readme.md]: ➡️ https://github.com/sindresorhus/gulp-rev/pull/267/commits/1d6480d2f95fd54a3dae53ea0c05665935962b85

My mistake, I thought that we could import the other packages of the examples using ES import

File: test/rev.js (hunk 3 above)
Comment (pioug) [reply test/rev.js:120]: ➡️ https://github.com/sindresorhus/gulp-rev/pull/267/commits/553466b0f75371af33a199c1f835e4ec6122124c

**Comment by sindresorhus** [comment]
Nice work 👍

**Pull request merged**